### DISCUSSION TITLE
Expose current import map in Deno as readonly

### DISCUSSION THREAD

**Discussion by hazae41** [denoland/deno#12575] [Ideas]
Related issue #12213 explains that we can't have a dynamic import map because of WICG standards

But can't we expose the current import map in Deno as readonly?

This would be a game changer for libraries that manage import maps:
- https://deno.land/x/importmap@0.2.0
- https://github.com/alephjs/aleph.js/issues/155

e.g. `Deno.imports`
```js
{ 
  path: "import_map.json",
  value: { 
    imports: { ... },
    scopes: { ... }
  }
}
```

**Comment by bartlomieju** [comment]
AFAIK there are no plans in the standard for introspection of import map and thus I'm not in favor of adding non-standard APIs for this purpose.

What are concrete use cases for such API?

**Reply by hazae41** [reply]
For example, at Aleph.js we are designing a tool to manage imports from the CLI.

Currently, it needs to either:
- use the default `import_map.json`
- use an extra configuration to define the import map url

`Deno.imports.url` would solve this, so we could just use the current import map

After that, we need to read it and parse it, so it requires read permission.
If the tool just wants to check updates, it would at least require read permission.

`Deno.imports.value` would solve this, so we don't need to ask for read permission.

We could also use imports to better analyse the code and dependency graph, issue warnings, polyfill, etc.

**Reply by bartlomieju** [reply]
Thanks for listing those. Still unless there's a specification allowing introspection I'm not in favor of adding a specialized API for that. I think having an explicit configuration option is the best way to solve this at the moment.
